app/controllers/pages/questions_controller.rb

```diff
@@ -6,34 +6,28 @@ def new
     is_optional = draft_question.is_optional == "true"
     page_heading = draft_question.page_heading
     guidance_markdown = draft_question.guidance_markdown
-    @question_form = Pages::QuestionForm.new(form_id: current_form.id, answer_type:, question_text:, answer_settings:, is_optional:, draft_question:)
+    @question_form = Pages::QuestionForm.new(answer_type:, question_text:, answer_settings:, is_optional:, draft_question:)
 
     @page = Page.new(form_id: current_form.id, question_text:, answer_type:, answer_settings:, is_optional:, page_heading:, guidance_markdown:)
     render :new, locals: { current_form: }
   end
 
   def create
-    @question_form = Pages::QuestionForm.new(page_params.merge(answer_settings: draft_question.answer_settings,
-                                                               page_heading: draft_question.page_heading,
-                                                               guidance_markdown: draft_question.guidance_markdown,
-                                                               draft_question:))
-    @page = Page.new(page_params.merge(answer_settings: draft_question.answer_settings,
-                                       page_heading: draft_question.page_heading,
-                                       guidance_markdown: draft_question.guidance_markdown,
-                                       answer_type: draft_question.answer_type))
+    @question_form = Pages::QuestionForm.new(page_params.merge(page_params_for_form_object))
+
+    @page = Page.new(page_params_for_forms_api)
 
     # TODO: Move Page creation to be part of the form submit method
     if @question_form.submit && @page.save
-      clear_questions_session_data
+      clear_draft_questions_data
       handle_submit_action
     else
       render :new, locals: { current_form: }, status: :unprocessable_entity
     end
   end
 
   def edit
-    @question_form = Pages::QuestionForm.new(form_id: current_form.id,
-                                             answer_type: draft_question.answer_type,
+    @question_form = Pages::QuestionForm.new(answer_type: draft_question.answer_type,
                                              question_text: page.question_text,
                                              hint_text: draft_question.hint_text,
                                              is_optional: draft_question.is_optional,
```

Comment:
Can we use `page_params_for_form_object` here?

Reply:
We cant use it because it includes params passed in from the form.  I do see what you mean that we could refactor this further so there is less duplication with passing in all these values to the forms object.